### ISSUE TITLE
Converting Types for JS API?

### ISSUE THREAD

**Issue by ScottPierce** [ForteScarlet/kotlin-suspend-transform-compiler-plugin#30]
There are certain types that JS doesn't work well with Kotlin i.e. if you have a function that returns a `List<String>`. It'd be nice if there was a way to automatically convert Kotlin collections to their correct JS counterparts.

List to Array
Map to JS Object
I'm sure there are others.

**Comment by ForteScarlet** [comment]
🤔️ This does seem to be a problem. Perhaps some way of providing support for type conversion of returned results (and which could be user-defined and not limited to JS) would be considered.
I haven't thought about it yet though, and the JS target is not available at the moment, so maybe I'll come back to it after [KT-53993](https://youtrack.jetbrains.com/issue/KT-53993) is fixed or #29 is implemented 😉 .

**Comment by ForteScarlet** [comment]
> This is no more a big issue. Collection types are now exportable.

Indeed. Since the [Kotlin 2.0.0 update](https://kotlinlang.org/docs/whatsnew20.html#improved-collection-interoperability), `List`, `Set` and `Map` can be exported.

Issue will be closed, if anyone has any new ideas, suggestions or has different opinions also feel free to add comments~

> BTW, if you want the synthesized bridge function to output arrays directly, you can also customize the transformer to do so—it's just a more cumbersome process with some limitations.
